hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -194,6 +202,47 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Set the ignition server availability condition by checking its deployment.
+	{
+		controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name)
+		deployment := ignitionserver.Deployment(controlPlaneNamespace.Name)
+		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(deployment), deployment); err != nil {
+			if apierrors.IsNotFound(err) {
+				meta.SetStatusCondition(&hcluster.Status.Conditions, metav1.Condition{
+					Type:               string(hyperv1.IgnitionServerAvailable),
+					Status:             metav1.ConditionFalse,
+					ObservedGeneration: hcluster.Generation,
+					Reason:             "IgnitionServerDeploymentNotFound",
+				})
+			} else {
+				return ctrl.Result{}, fmt.Errorf("failed to get ignition server deployment: %w", err)
+			}
+		} else {
+			available := false
+			for _, cond := range deployment.Status.Conditions {
```

Comment:
It'd be nice to try to simplify this a bit and reduce the nested if/else logic in this stanza. How about:

```
		const IgnitionServerDeploymentNotFoundReason = "IgnitionServerDeploymentNotFound"
		const IgnitionServerDeploymentNotAvailableReason = "IgnitionServerDeploymentNotAvailable"
		const IgnitionServerDeploymentAvailableReason = "IgnitionServerDeploymentAvailable"

		// Set IgnitionServerAvailable=false unless proven otherwise
		meta.SetStatusCondition(&hcluster.Status.Conditions, metav1.Condition{
			Type:               string(hyperv1.IgnitionServerAvailable),
			Status:             metav1.ConditionFalse,
			ObservedGeneration: hcluster.Generation,
			Reason:             IgnitionServerDeploymentNotAvailableReason,
		})
		controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name)
		deployment := ignitionserver.Deployment(controlPlaneNamespace.Name)
		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(deployment), deployment); err != nil {
			if !apierrors.IsNotFound(err) {
				return ctrl.Result{}, fmt.Errorf("failed to get ignition server deployment: %w", err)
			}

			if apierrors.IsNotFound(err) {
				meta.SetStatusCondition(&hcluster.Status.Conditions, metav1.Condition{
					Type:               string(hyperv1.IgnitionServerAvailable),
					Status:             metav1.ConditionFalse,
					ObservedGeneration: hcluster.Generation,
					Reason:             IgnitionServerDeploymentNotFoundReason,
				})
			}
		}

		for _, cond := range deployment.Status.Conditions {
			if cond.Type == appsv1.DeploymentAvailable && cond.Status == corev1.ConditionTrue {
				meta.SetStatusCondition(&hcluster.Status.Conditions, metav1.Condition{
					Type:               string(hyperv1.IgnitionServerAvailable),
					Status:             metav1.ConditionTrue,
					ObservedGeneration: hcluster.Generation,
					Reason:             IgnitionServerDeploymentAvailableReason,
				})
				break
			}
		}
	}
```

Reply:
`deployment` really isn't that safe to use unless `err == nil`, but your idea to pessimistically set status allowed for some nice cleanup here. I refactored to use that idea and extracted a nearby inline function to minimize nesting and LOC overall, how does it look now?